### PR TITLE
Create PT-BR translation

### PR DESCRIPTION
**Pull request by LeaoSomogyi**
* Change `reasons` structure to separate by circunstance, in order to make easier to split by language.
* Create a new `Language` component
* Include `language` on `props`
* Create methods to translate some html content
* Include `language` on both `api` and `api/slack`

image

Sent by query param as tz is:

image

### CODE REVIEW

**Comment by LeaoSomogyi** [comment]
@baires running local doesn't have any error, but vercel is on failed state to deploy on preview environment. Did i make something wrong?

File: component/widget.js [hunk 1]
```diff
@@ -52,7 +60,28 @@ export default class Widget extends React.Component {
    * @return string[]
    */
   getReasons() {
-    return dayHelper(this.props.now)
+    return dayHelper(this.props.now, this.props.language)
+  }
+
+  getHitTextTranslation() {
+    if (this.props.language === 'pt-br')
+      return 'Aperte';
```
Comment (Ronkiro) [inline component/widget.js:68]: Not sure if this code would be visually good when the amount of languages increases. I would suggest using a library (I'm not the best to suggest libraries for react, but maybe https://www.npmjs.com/package/react-i18next)

**Review by Ronkiro — CHANGES_REQUESTED** [review]
Great work!!! I suggested some changes, give it a check!! :D

File: README.md [hunk 2]
```diff
@@ -13,7 +13,11 @@
 
 ## Add your reasons
 
-Reasons are located under [reasons.js](https://github.com/baires/shouldideploy/blob/master/helpers/reasons.js)
+Reasons are located under [reasons](https://github.com/baires/shouldideploy/blob/master/helpers/reasons) folder. Files are separated by circunstance.
+
+There's two languages supported: English and Brazilian Portuguese.
```
[the remark this answers is not in the document]
Comment (LeaoSomogyi) [reply README.md:18]: @Ronkiro i agree with you, gonna delete this line :sweat_smile:

File: README.md [hunk 3]
```diff
@@ -13,7 +13,11 @@
...
+
+Choose which language you want to add your reason and go ahead!
```
[the remark this answers is not in the document]
Comment (LeaoSomogyi) [reply README.md:20]: @Ronkiro vi que você é brasileiro também, então vou responder em pt-br hahaha nesse contexto, `which` fica com o sentido de qual, então meio que seria "Escolha qual idioma você deseja adicionar uma razão e vai em frente!"

File: component/widget.js (hunk 1 above)
Comment (LeaoSomogyi) [reply component/widget.js:68]: @Ronkiro dei uma pesquisada nesse pacote e acho que vai dar mais trabalho configurá-lo do que implementar alguma outra solução manual. Estou com a ideia de criar um json com os textos da página traduzidos e carregá-los quando a página iniciar. O que acha?

**Comment by LeaoSomogyi** [comment]
@Ronkiro changes applied :smile:

File: component/widget.js (hunk 1 above)
Comment (Ronkiro) [reply component/widget.js]: Acho uma boa! É suficiente, eu acho

**Review by Ronkiro — APPROVED** [review]
Nice!!

**Comment by baires** [comment]
Tx for the hard work here. I think we need to put this on hold til we we complete #718 (help needed BTW)

**Comment by LeaoSomogyi** [comment]
> Tx for the hard work here. I think we need to put this on hold til we we complete #718 (help needed BTW)

@baires why not merge it and then start the migration to `ts`? I'm really wanna use the API on my pipelines, but with `PT-BR` messages :sweat_smile: 

BTW, i could help with the issue as well

**Comment by dennissiq** [comment]
Nice Job Bro!!

**Comment by baires** [comment]
@LeaoSomogy @Ronkiro  now that typescript migration its done, its ok to start working again on this feature

**Comment by LeaoSomogyi** [comment]
> @LeaoSomogy @Ronkiro now that typescript migration its done, its ok to start working again on this feature

Done!

**Comment by baires** [comment]
I need to review this implementation deeper but:
- Make sure to run `npm run lint`
- Footer text should be also translated
- `/?lang=pt-br` should render pt language not english

I think the best way to tackle this feature is to use Next.js internationalized [routing](https://nextjs.org/docs/advanced-features/i18n-routing) since v10.0.0.

```
module.exports = {
  i18n: {
    locales: ['en', 'pt', 'es'],
    defaultLocale: 'en',
  },
}
```

```
import { useTranslation } from 'next-translate'

export default function Home() {
  const { t } = useTranslation('home')

  return (
    <div>
      <h1>{t('welcome')}</h1>
    </div>
  )
}
```

**Review by baires — CHANGES_REQUESTED** [review]
Lets make sure to use, there 
https://nextjs.org/docs/advanced-features/i18n-routing

**Comment by LeaoSomogyi** [comment]
> Lets make sure to use, there https://nextjs.org/docs/advanced-features/i18n-routing

@baires you mean to use this on html texts right?

**Comment by Ronkiro** [comment]
Yes @LeaoSomogyi, in every place that will be in portuguese for the end-user. This will allow the translation to easily scale to other languages while following common patterns. Your code is very nice rn, but i think it would be a good idea too (as it's native from next.js)

**Review by arieldossantos — APPROVED** [review]

**Pull request closed without merging**